### PR TITLE
Job fixes, unit test fixes.

### PR DESCRIPTION
**Pull request by seeker25**

### CODE REVIEW

File: jobs/payment-jobs/tasks/cfs_create_invoice_task.py [hunk 1]
```diff
@@ -291,17 +291,15 @@ def _create_pad_invoices(cls):  # pylint: disable=too-many-locals
             mailer.publish_mailer_events('pad.invoiceCreated', payment_account, additional_params)
             # Iterate invoice and create invoice reference records
             for invoice in account_invoices:
-                # Create invoice reference, payment record and a payment transaction
-                InvoiceReference.create(
+                invoice_reference = InvoiceReferenceModel(
```
Comment (seeker25) [inline jobs/payment-jobs/tasks/cfs_create_invoice_task.py:294]: Creating invoice references this way takes 1 second at a time. I ran into a backup where we had 20,000 invoices to create.  20,000 invoice references / 60 seconds * 60 minutes = 5.5 hours to create all these invoice references. 

This reduced the time to approx 3-4 minutes.

File: jobs/payment-jobs/tasks/unpaid_invoice_notify_task.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def _notify_for_ob(cls):  # pylint: disable=too-many-locals
                 cfs_account = CfsAccountModel.find_effective_by_account_id(payment_account_id)
 
                 # emit account mailer event
-                addition_params_to_mailer = {'transactionAmount': total[0][0],
+                addition_params_to_mailer = {'transactionAmount': float(total[0][0]),
```
Comment (seeker25) [inline jobs/payment-jobs/tasks/unpaid_invoice_notify_task.py:75]: Can't serialize Decimal error.

File: pay-api/src/pay_api/services/paybc_service.py [hunk 3]
```diff
@@ -146,43 +145,6 @@ def _get_receipt_by_number(self, access_token: str = None, receipt_url: str = No
...
-            ]
-        )
-
-        adjustment_response = self.post(adjustment_url, access_token, AuthHeaderType.BEARER, ContentType.JSON,
-                                        adjustment)
-
-        current_app.logger.debug('>Created PayBC Invoice Adjustment')
-        return adjustment_response.json()
-
-    def _get_invoice(self, payment_account: PaymentAccount, inv_number: str, access_token: str):
```
Comment (seeker25) [inline pay-api/src/pay_api/services/paybc_service.py:175]: Unused code

File: pay-api/src/pay_api/services/paybc_service.py [hunk 4]
```diff
@@ -146,43 +145,6 @@ def _get_receipt_by_number(self, access_token: str = None, receipt_url: str = No
         receipt_url = receipt_url + f'{receipt_number}/'
         return self.get(receipt_url, access_token, AuthHeaderType.BEARER, ContentType.JSON, True).json()
 
-    def _add_adjustment(self, payment_account: PaymentAccount,  # pylint: disable=too-many-arguments
```
Comment (seeker25) [inline pay-api/src/pay_api/services/paybc_service.py:149]: Unused code

**Comment by Jxio** [comment]
looks good, thanks.

File: pay-api/src/pay_api/services/payment_account.py [hunk 5]
```diff
@@ -538,12 +538,8 @@ def find_by_auth_account_id(cls, auth_account_id: str) -> PaymentAccount:
     @classmethod
     def find_by_id(cls, account_id: int):
         """Find pay account by id."""
-        current_app.logger.debug('<find_by_id')
```
Comment (seeker25) [inline pay-api/src/pay_api/services/payment_account.py:541]: Spams the logs, don't really need this.

**Review by Jxio — APPROVED** [review]

**Pull request merged**